gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -131,17 +136,43 @@
     // into the repo?
     this.mapFeatures =
         LiveDataReactiveStreams.fromPublisher(
-            projectRepository
-                .getActiveProject()
-                .switchMap(this::getFeaturesStream)
-                .map(MapContainerViewModel::toMapFeatures));
+            Flowable.combineLatest(
+                projectRepository
+                    .getActiveProject()
+                    .switchMap(this::getFeaturesStream)
+                    .map(MapContainerViewModel::toMapFeatures),
+                selectedFeature,
+                this::updateSelectedFeature));
     this.mbtilesFilePaths =
         LiveDataReactiveStreams.fromPublisher(
             offlineBaseMapRepository
                 .getDownloadedTileSourcesOnceAndStream()
                 .map(set -> stream(set).map(TileSource::getPath).collect(toImmutableSet())));
   }
 
+  private ImmutableSet<MapFeature> updateSelectedFeature(
+      ImmutableSet<MapFeature> features, Optional<Feature> selectedFeature) {
+    Timber.v("Updating selected feature style");
+    if (selectedFeature.isEmpty()) {
+      return features;
+    }
+    ImmutableSet.Builder updatedFeatures = ImmutableSet.builder();
+    String selectedFeatureId = selectedFeature.get().getId();
+    for (MapFeature feature : features) {
```

Comment:
It's not clear to me why we need to iterate here. Why wouldn't we just pass the feature directly to the VM via the click handler using an RX stream? e.g., if we pass via `selectedFeature`, wouldn't we just handle such a side-effect in a subscription?

```
selectedFeature.onNext(feat -> if isinstanceOf JSON setStroke.......etc.)
```

The only reason we maybe wouldn't want to do that is to avoid using old values (updated happened on the server somewhere so the repo contains new data post mutation), but this doesn't solve that case either, as to get around that we'd want to request a fresh copy from the DB, right?

In other words, wouldn't a simpler model leveraging RX be:

MapAdapter[onClick { featureProcessor.OnNext(someFeature)}] -> MapFragment[subscribe(Adapter.featureProcessor).onNext(change styles).emitToVMconnectingStream] -> VM.vmConnectingStream.onNext(do something w/ feature data)

or something like that

Reply:
Each emission in the stream of features coming from the data store contains the full set of features to show on the map. We're effectively trying to modify one item in that stream if it's selected; the `updateSelectedFeature` function does this for us. There isn't a stream of individual features to modify, so emitting a single value wouldn't help us here (unless I'm misunderstanding something, which is entirely possible!)

Reply:
@scolsen Since we spoke earlier, is it ok to proceed to merge? If so can I get a restamp?